### ISSUE TITLE
"env use" fails with "NameError: name 'PosixPath' is not defined"

### ISSUE THREAD

**Issue by elferia** [python-poetry/poetry#2711]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 18.04.4 LTS
- **Poetry version**: 1.0.10

## Issue
When I try the following command,

```
poetry env use 3.8
```

it will fail with the following message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'PosixPath' is not defined
```

I guess this is caused by the following line.
https://github.com/python-poetry/poetry/blob/d3c9049a18ae33baacfcb5c698777282f2f58128/poetry/utils/env.py#L625
Variable `venv` is a `Path` object, and it is finally passed to `str.format()`
https://github.com/python-poetry/poetry/blob/d3c9049a18ae33baacfcb5c698777282f2f58128/poetry/utils/env.py#L672
Then `CREATE_VENV_COMMAND` is
https://github.com/python-poetry/poetry/blob/d3c9049a18ae33baacfcb5c698777282f2f58128/poetry/utils/env.py#L105-L106

```
$ python -c 'import pathlib; print("{!r}".format(pathlib.Path("/")))'
PosixPath('/')
```

So the interpreter will first read a line like `path = PosixPath('...')` but `PosixPath` is not defined.

I think this bug is introduced in 1.0.10 https://github.com/python-poetry/poetry/commit/54870697d921ae6efdbba0bc86f6b54c86d6896a

I guess it will be fixed by replacing `venv` to `str(venv)` so I tried to fix, but the commit above is replacing `str(venv)` to `venv` and I can't understand the purpose to do so. I can't fix it.

Thanks.

**Comment by elferia** [comment]
version 1.1.0a3 not include 5487069 https://github.com/python-poetry/poetry/blob/a50d87dbdcd539499ce7e5c7e9149a491505af52/poetry/utils/env.py#L664
So workaround is installing preview version.

**Comment by abn** [comment]
@elferia appreciate the bug report and the analysis. As you have already identified, the preview version will resolve this. And since we recently released the last `1.0.z` release and next release is expected soon, resolving this.

**Comment by alexlouden** [comment]
I had the same issue, upgrading to the latest preview version fixed it for me - thanks1

```
$ poetry self update --preview
Updating to 1.1.0b2
```

**Comment by tony** [comment]
Thanks @alexlouden 

I had to do : `pip install --user --pre poetry -U`

Since my poetry installation gave me this warning: `Poetry was not installed with the recommended installer. Cannot update automatically.`

**Comment by goodcheer** [comment]
Thanks @tony 

if you're using `pipx` as install manager, you can get pre-release by : `pipx install --pip-args='--pre' poetry` (confimred working)

or upgrade exisiting one `pipx install --pip-args='--pre -U' poetry` (haven't tried this one, but might work)

**Comment by jrgoldfinemiddleton** [comment]
> pipx install --pip-args='--pre -U' poetry

I had to use this instead:
`pipx upgrade --pip-args='--pre' poetry`

**Comment by gavinbeatty** [comment]
Still running into this issue when doing `poetry shell` on mac latest stable, works on preview 1.1.0rc1.

**Comment by abn** [comment]
@gavinbeatty https://github.com/python-poetry/poetry/issues/2711#issuecomment-663499610

**Comment by languitar** [comment]
Isn't there a chance to get a patch release for this bug for the 1.0 line? Seems a few dependencies still block upgrading things like the archlinux package and user there are then stuck with this broken version.

**Comment by tony** [comment]
@languitar Can you clarify "1.0 line"? (What is your `tmuxp -V`?)

> Isn't there a chance to get a patch release for this bug for the 1.0 line? Seems a few dependencies still block upgrading things like the archlinux package and user there are then stuck with this broken version.

Can you provide more information?

**Comment by tony** [comment]
@abn Hi there!  I got a a question (echo'ing) @languitar at https://github.com/python-poetry/poetry/issues/2711#issuecomment-756128303

Is there any way around this upgrading to 1.1 (I can't do that - I have projects depending on 1.0 and it's the team's choice whether poetry upgrades, the lockfiles aren't compatible to my knowledge between 1.0 and 1.1)

Is this something that could be backported into 1.0? (I am on 1.0.10)

What was the fix for this in 1.1?  Is it a trivial one?  

I don't believe it's solved by https://github.com/python-poetry/poetry/commit/54870697d921ae6efdbba0bc86f6b54c86d6896a, which is in 1.0.10, since I still get the issue.  The PR at #2742 doesn't specify what change fixed it in 1.1 at the time

Any workarounds aside from upgrading anyone know of? This sorta breaks my setup.
